src/domain/payments/onchain-payment-flow-builder.ts

```diff
@@ -187,8 +181,14 @@ const OPFBWithAmount = <S extends WalletCurrency, R extends WalletCurrency>(
         const updatedStateFromBtcProposedAmount = async (
           btcProposedAmount: BtcPaymentAmount,
         ): Promise<OPFBWithConversionState<S, R> | DealerPriceServiceError> => {
-          const convertedAmount = await mid.usdFromBtc(btcProposedAmount)
-          if (convertedAmount instanceof Error) return convertedAmount
+          const convertedAmountRaw = await mid.usdFromBtc(btcProposedAmount)
+          if (convertedAmountRaw instanceof Error) return convertedAmountRaw
+
+          // Round up USD if sats is non-zero in intraledger (mid-price) case and cents is zero
```

Comment:
Note on this, it's only to not fail in the edge-case that one user sends 1 sat from their own **usd** wallet to another user's **usd** wallet

![image](https://user-images.githubusercontent.com/17693119/230019453-aba20247-2d2a-4821-a651-11bcb1bf5f13.png)